### ISSUE TITLE
Support for Windows PowerShell 3 and 4

### ISSUE THREAD

**Issue by tylerl0706** [PowerShell/Polaris#64]
As it turns out:

`[PolarisCore.Polaris]::new()` is PowerShell 5 syntax.

We should use a syntax that's more supported.

Maybe something like:
`New-Object PolarisCore.Polaris` if possible.

**Comment by affieuk** [comment]
Should have said I tried `New-Object PolarisCore.Polaris`, it doesn't work

**Comment by tylerl0706** [comment]
Thanks for that. If you are aware of a working syntax, let me know. I'll keep researching.

**Comment by tylerl0706** [comment]
@affieuk can you try New-Object -TypeName PolarisCore.Polaris -ArgumentList { param($str) Write-Verbose "$str" }

**Comment by affieuk** [comment]
Here's the error
```PowerShell
PS C:\Polaris> Import-Module .\Polaris.psm1
PS C:\Polaris> New-Object PolarisCore.Polaris
New-Object : A constructor was not found. Cannot find an appropriate constructor for type PolarisCore.Polaris.
At line:1 char:1
+ New-Object PolarisCore.Polaris
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (:) [New-Object], PSArgumentException
    + FullyQualifiedErrorId : CannotFindAppropriateCtor,Microsoft.PowerShell.Commands.NewObjectCommand
```

However I've figured out the correct syntax
`New-Object PolarisCore.Polaris -ArgumentList $logger`

**Comment by affieuk** [comment]
I've changing line 581 to `$global:Polaris = New-Object -TypeName PolarisCore.Polaris -ArgumentList $logger` in Polaris.psm1, I can confirm that Polaris functions on PSv4. Here's the output from `invoke-pester`
```PowerShell
PS C:\Polaris\test> Invoke-Pester
Executing all tests in '.'

Executing script C:\Polaris\test\PolarisMiddleware.Tests.ps1

  Describing Test middleware creation/usage

    Context Test -UseJsonBodyParserMiddleware flag
      [+] Adds the middleware to the list of middlewares 1.12s
      [+] Can use the body parameters in route scripts 252ms

    Context Test adding a new middleware
      [+] Adds the middleware to the list of middlewares 163ms
      [+] Manipulates the Request before the Route script gets it 66ms

Executing script C:\Polaris\test\PolarisRoute.Tests.ps1

  Describing Test route creation

    Context Using New-WebRoute
      [+] with -Path '/test' -Method 'GET' -ScriptPath '' -ScriptBlock '
    $response.Send("test script");
' 4.29s
      [+] with -Path '/test' -Method 'POST' -ScriptPath '' -ScriptBlock '
    $response.Send("test script");
' 17ms
      [+] with -Path '/test' -Method 'PUT' -ScriptPath '' -ScriptBlock '
    $response.Send("test script");
' 17ms
      [+] with -Path '/test' -Method 'DELETE' -ScriptPath '' -ScriptBlock '
    $response.Send("test script");
' 49ms
      [+] with -Path '/test' -Method 'GET' -ScriptPath './test.ps1' -ScriptBlock '' 28ms
      [+] with -Path '/test' -Method 'POST' -ScriptPath './test.ps1' -ScriptBlock '' 22ms
      [+] with -Path '/test' -Method 'PUT' -ScriptPath './test.ps1' -ScriptBlock '' 22ms
      [+] with -Path '/test' -Method 'DELETE' -ScriptPath './test.ps1' -ScriptBlock '' 23ms

    Context Using New-GetRoute
      [+] with -Path '/test' -ScriptPath './test.ps1' -ScriptBlock '' 127ms
      [+] with -Path '/test' -ScriptPath '' -ScriptBlock '
    $response.Send("test script");
' 26ms

    Context Using New-PostRoute
      [+] with -Path '/test' -ScriptPath './test.ps1' -ScriptBlock '' 46ms
      [+] with -Path '/test' -ScriptPath '' -ScriptBlock '
    $response.Send("test script");
' 74ms

    Context Using New-PutRoute
      [+] with -Path '/test' -ScriptPath './test.ps1' -ScriptBlock '' 53ms
      [+] with -Path '/test' -ScriptPath '' -ScriptBlock '
    $response.Send("test script");
' 24ms

    Context Using New-DeleteRoute
      [+] with -Path '/test' -ScriptPath './test.ps1' -ScriptBlock '' 53ms
      [+] with -Path '/test' -ScriptPath '' -ScriptBlock '
    $response.Send("test script");
' 18ms

    Context Using New-StaticRoute
      [+] Will create a route for every file in a directory without the RoutePath param 96ms
      [+] Will create a route for every file in a directory with the RoutePath param 54ms

    Context Using Get-WebRoute and Remove-WebRoute
      [+] Will get the object with the routes 65ms
      [+] will remove the routes 35ms

Executing script C:\Polaris\test\PolarisServer.Tests.ps1

  Describing Test webserver use

    Context Test different route responses
      [+] test /helloworld route 394ms
      [+] test /helloworld route with query params that do nothing 313ms
      [+] test /hellome route with query param 43ms
      [+] test /hellome route without query param 35ms
      [+] test /wow route 195ms
      [+] test /example route 49ms
      [+] test /public/index.html static route 65ms

    Context Test starting and stopping of the server
      [+] Can properly shut down the server 12.44s
Tests completed in 20.3s
Tests Passed: 32, Failed: 0, Skipped: 0, Pending: 0, Inconclusive: 0
PS C:\Polaris\test> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      4.0
WSManStackVersion              3.0
SerializationVersion           1.1.0.1
CLRVersion                     4.0.30319.34011
BuildVersion                   6.3.9600.16394
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0}
PSRemotingProtocolVersion      2.2
```

**Comment by tylerl0706** [comment]
@affieuk  If you would like to submit a PR for that 1 line on its own, I would accept it :) 

If not, I can do it later today.

**Comment by affieuk** [comment]
Sure, I'll do that

**Comment by tylerl0706** [comment]
Fixed in #65 thanks to @affieuk!
